### ISSUE TITLE
Weird behaviour while monitoring zendesk queue

### ISSUE THREAD

**Issue by BenjaminSzeszler** [facetoe/zenpy#634]
Hi Everyone,

I wrote a method that pulls the last 10 tickets from zendesk. Here is a dummy version of my code:

```
from zenpy import Zenpy
import time

class ZendeskApiHandler:
    def __init__(self, zendesk_creds):
        self.zenpy_client = Zenpy(**zendesk_creds)

    def get_last_10_tickets(self):

        params = {'type': 'ticket', 'sort_by': 'created_at', 'sort_order': 'desc'}

        tickets = self.zenpy_client.search(**params)

        ticket_generator = [ticket for ticket in tickets]

        return ticket_generator[:10]

if __name__ == "__main__":

    zendesk_api_handler = ZendeskApiHandler(zendesk_creds = {
             "email": "-------",
             "token": "-------",
             "subdomain": "------"
        })
    
    while True:

        last_10_tickets = zendesk_api_handler.get_last_10_tickets()

        print(last_10_tickets)

        time.sleep(1)
```

When I ran the script I noticed the following weird behavior:

- At first it worked prefectly. Output:
```
[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]
```
- Then I created a new ticket in Zendesk and it got weird. Output:
```
[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21), Ticket(id=22)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]

[Ticket(id=31), Ticket(id=30), Ticket(id=29), Ticket(id=28), Ticket(id=27), Ticket(id=26), Ticket(id=25), Ticket(id=24), Ticket(id=23), Ticket(id=21)]
...
```
Do you have an idea what is the reason for this?
Thanks for this great repo by the way <33

**Comment by Ptr314** [comment]
Hi @BenjaminSzeszler!

A I know, Zendesk search API doesn't use the database directly, it uses some kind of cache, whith has a delay for getting actual data. In our projects we take it to account and when we create something in Zendesk which we need back in a short time, we use our own cache and check it first before calling Search API. These cache entries live about a minute, but it guarantees what we don't create something twice because of Zendesk Search data actuality delays.

Zenpy also has a cache with its own TTL, and your first results may be taken from there.

So, IMHO, your case is not a problem with Zenpy itself and cannot be easily solved.

**Comment by BenjaminSzeszler** [comment]
Hi @Ptr314 

Thanks for your answer <3

Since I opened the issue I tested the Zendesk API directly and the issue occurs that way too, so you’re absolutely right. The issue is not with Zenpy. 

My use case is that I want to monitor zendesk and get notified when a new ticket is created, but I don’t want to host a webserver for a webhook. I came up with a solution for this issue and now it works fine but unfortunately the delay is still there. I guess there is nothing I can do about that. Right?

Thanks for your answer again anyways!
